dip-template/pallets/nft-marketplace/src/lib.rs

```diff
@@ -250,30 +241,25 @@ pub mod pallet {
 		/// The maximum length of data stored in for post codes.
 		#[pallet::constant]
 		type PostcodeLimit: Get<u32>;
+
+		type DidIdentifier: Parameter + MaxEncodedLen;
+		type BuyTokenOrigin: EnsureOrigin<Self::RuntimeOrigin, Success = Self::DidIdentifier>;
```

Comment:
This will most likely be a more generic origin rather than an origin for a specific call. What's important is the trait it's required to implement, and the expected `Success` type, which will be a Kilt DID.